packages/guides-restructured-text/src/RestructuredText/Toc/ToctreeBuilder.php

```diff
@@ -4,39 +4,53 @@
 
 namespace phpDocumentor\Guides\RestructuredText\Toc;
 
+use phpDocumentor\Guides\Nodes\Menu\ParsedMenuEntryNode;
 use phpDocumentor\Guides\ParserContext;
+use phpDocumentor\Guides\RestructuredText\Parser\EmbeddedUriParser;
 use phpDocumentor\Guides\RestructuredText\Parser\LinesIterator;
 
 use function array_filter;
 use function array_map;
 
 class ToctreeBuilder
 {
+    use EmbeddedUriParser;
+
     /**
      * @param mixed[] $options
      *
-     * @return string[]
+     * @return ParsedMenuEntryNode[]
      */
-    public function buildToctreeFiles(
+    public function buildToctreeEntries(
         ParserContext $parserContext,
         LinesIterator $lines,
         array $options,
     ): array {
-        $toctreeFiles = [];
+        $toctreeEntries = [];
 
-        foreach ($this->parseToctreeFiles($lines) as $file) {
-            $toctreeFiles[] = $file;
+        foreach ($this->parseToctreeEntryLines($lines) as $entry) {
+            $toctreeEntries[] = $entry;
         }
 
-        return $toctreeFiles;
+        return $toctreeEntries;
     }
 
-    /** @return string[] */
-    private function parseToctreeFiles(LinesIterator $lines): array
+    /** @return ParsedMenuEntryNode[] */
+    private function parseToctreeEntryLines(LinesIterator $lines): array
     {
-        return array_filter(
+        $linesArray =  array_filter(
             array_map('trim', $lines->toArray()),
             static fn (string $file): bool => $file !== '',
         );
+
+        $result = [];
+        foreach ($linesArray as $line) {
+            $parsed = $this->extractEmbeddedUri($line);
+            $title = $parsed['text'];
+            $reference = $parsed['uri'];
+            $result[] = new ParsedMenuEntryNode($reference, $title);
```

Comment:
I would like to see this as inline construction, this will save some memory as php is coping all strings. 
```suggestion
            $parsed = $this->extractEmbeddedUri($line);
            $result[] = new ParsedMenuEntryNode($parsed['uri'], $parsed['text']);
```

Reply:
yeah, this was a left-over